task.go

```diff
@@ -29,13 +35,26 @@ func (t Task) GetInterval() string {
 
 // Zeros the parameters that may change in a run()
 func (t *Task) Zero() {
+	t.MessageCountLock.Lock()
```

Comment:
It's not clear to me why you need so many locks.  The locking seems to be very granular.  Having just one mutex to protect one struct is good enough.